### PR TITLE
AI Eyebot rework

### PR DESCRIPTION
**Pull request by Paaiii**
[SILICONS] [REWORK]

### **TO PUT IT ALL VERY QUICKLY:**
1. Updates the drone HUD to not be terrible
2. Gives drones watered-down babyproofed versions of modules
3. Added more in-depth functionality for drones to cyborg docking stations
4. Gives drones an access panel similar to cyborgs for replacing interface/module/cells
5. Resprites eyebots.

https://github.com/goonstation/goonstation/assets/124699693/728e52ea-518f-452c-a77b-d02584309b89

### **To put it all much, much less quickly:**
Replaces the janky old hivebot mob with a fancy new drone mob.
drones have the same base functionality as the old hivebots, however:

### 1. They have a HUD that isn’t old, clunky, or missing vital information, following in line with the existing cyborg HUD

1a-newUI

This includes:
a) Actually displaying the health of the drone in some way so you aren’t shocked when it explodes into bits due to a terminal case of having a max HP of 25,

b) An indicator for if the drone is pulling an object, so that you don’t need to ctrlclick on the item again to stop pulling it,

c) A targeter so that drones can actually target head/limbs/whatnot,

d) A radio menu– you know, I could list all of these separately but I’ll just- O2 display, temp display, intent switch option, battery display that shows percentage– you get the point.

### 2. They have actual modules so they can do stuff that isn't just moving objects from point A to B for the AI.

2a-modules

**_PLEASE NOTE--_ AI drones at roundstart get a default module with only the tools existing eyebots get. This default module cannot be swapped at docking stations or manually. The drone modules are only usable with drones assembled during the round.**

These modules are SUBSTANTIALLY watered down versions of existing cyborg modules.
Here's a full list for you to scrutinize:
```
Civilian (pulls from civilian, brocop):
    - Extinguisher
    - Spray bottle
    - Sponge
    - Pitcher
    - Pen
    - Cardboard tube

Engineering (pulls from engineering, mining):
    - Extinguisher
    - Welding tool
    - Device analyzer
    - Soldering iron
    - Empty steel sheet stack
    - Cargo transporter

Medsci (pulls from medical, science):
    - Drone robospray (gets ephedrine, atropine, salbutamol, and saline)
    - Suture
    - Syringe
    - Beaker (small)
    - Hand labeler
    - GPS
```

It’s worth mentioning before someone gets clever about it that, no, you can’t put regular cyborg modules into the drones. It doesn’t accept them. Nor do cyborgs accept the drone modules.

The UI for the module rewriter has been updated to account for the new variant of module:
2b-modulerewriter

*(They also only get two module slots, as opposed to three. This is because they don't have arms, and as such them having the same amount of module slots as normal cyborgs without any way to lose those slots seems silly.)*

### 3. – Drones have an access panel for maintenance, similar to cyborgs.

3a-accesspanel
*fig. 1 - An eyebot drone with its noggin opened and ripe for the picking.*

This panel can be opened by swiping an ID with robotics access, then prying it open with a crowbar. Similar to cyborgs, as I said.

This panel allows you to swap out the drones cell, module, and AI interface board.

### 5. – Docking stations actually work with drones. They didn’t with eyebots.

5a-dockingstation

This includes:
	a) The ability to repair damages in a docking station
	b) The ability to swap power cells and modules in the docking station,
	c) And most importantly, the ability to equip a stupid little cosmetic.

5b-stupidhat
*Pictured above: Stupid little cosmetic.*

*(Just two for now, but functionality is in place for more to be trivial to add.)*

### 6. – Construction is slightly different than the old eyebots.

6a-gpcsconst
*Fig. 2 - A gpcs drone at various stages of assembly.*

To construct an eyebot it was:
1. add plating
2. add wires
3. add cell
4. add interface
5. add radio
7. wrench

To construct a drone:
1. add plating
2. add wires
3. add radio
4. add cell and interface (both of which are removable from frame with a wrench, and will inherit their type to the resultant drone)
5. wrench
6b-frame
So, basically, makes it work more like cyborg frames. Who would have guessed.

## Why's this needed? 
 - Existing eyebots are kind of bad - and not in an intended way, like they’re meant to be bad, but bad as in they’re lacking essential QoL stuff (health indicator, essential UI elements, docker functionality)

- Adding lite versions of modules for drones will hopefully wean AI players off of cyborg shells enough for me to be able to pitch removing the AIs ability to have cyborg shells
`(Why do they have those. The fact that the AI can also at will take control of a fully-functional cyborg essentially makes the AI into cyborg++ with no literally downsides, and discourages cooperation between the AI and cyborg players because the AI can just. do anything a cyborg could and doesn’t need to talk to the cyborgs. It’s dumb. It’s also besides the point. so MOVING ON)`

- um

- it’s my birthday tomorrow and i think that legally means you have to merge it (/j)

- Oh yeah also more customization more good

## Changelog 

```changelog
(u)Paaiii
(*)Overhauled how AI shells work! See the PR for more details.
(+)Drones now can be built to have lite versions of modules.
(+)Drones got a fancy resprite to go with the other changes.
(+)Drone UI updated.
(+)Drones have an access panel which allows the easy replacement of interface, cell, or module.
```

### CODE REVIEW

**Comment by Paaiii** [comment]
OH BOY why did it request a review from literally everyone. Whoopsies. Sorry about that

**Comment by DisturbHerb** [comment]
I am personally against providing modules and more tools to the eyebot, it seems to me that it could devalue the use of AI interfaced borgs. I also recognise that that might be an overwhelmingly unpopular opinion.

**Comment by DisturbHerb** [comment]
> OH BOY why did it request a review from literally everyone. Whoopsies. Sorry about that

they own da code das why :iiam:

**Review by ZeWaka — CHANGES_REQUESTED** [review]
Remove the old code if it's not going to be used

**Comment by Paaiii** [comment]
> I am personally against providing modules and more tools to the eyebot, it seems to me that it could devalue the use of AI interfaced borgs. I also recognise that that might be an overwhelmingly unpopular opinion.

HONESTLY... devaluing interfaced borgs is kind of the point here hsdjkdsgh
i'm anti-aiborgshell it feels weird to me that AIs get to be another job in entirety on top of being the AI. I think if AIs couldn't use borgs as shells it would be a lot more interesting because it would require the AI to coordinate with cyborgs for more complex physical tasks, rather than the current situation where the AI is just... cyborg plus plus. Cyborg but better.

Don't tell anyone but my evil master plan is to devalue interfaced borgs enough to where I can convince everyone it would be best to remove them, leaving only drone shells

PLUS, if that doesn't help any, the modules the drones get are pretty basic- there's a lot of things they just Can't do that cyborg modules can

**Comment by wolfolotl** [comment]
> I am personally against providing modules and more tools to the eyebot, it seems to me that it could devalue the use of AI interfaced borgs. I also recognise that that might be an overwhelmingly unpopular opinion.

seconded, it feels like this will basically make roboticists obsolete except for being The Guys Who Print Blank Modules from a silicon perspective. additionally, I am not a fan of how beefy these modules are in terms of functionality, with the medical module basically still allowing them to keep pace with a human doctor in every area except surgery. I get the desire to make some things More Useful, but from a design perspective, I feel like these things SHOULD be pretty useless in a lot of regards

**Comment by Glamurio** [comment]
> HONESTLY... devaluing interfaced borgs is kind of the point here hsdjkdsgh i'm anti-aiborgshell it feels weird to me that AIs get to be another job in entirety on top of being the AI. I think if AIs couldn't use borgs as shells it would be a lot more interesting because it would require the AI to coordinate with cyborgs for more complex physical tasks, rather than the current situation where the AI is just... cyborg plus plus. Cyborg but better.
> 
> Don't tell anyone but my evil master plan is to devalue interfaced borgs enough to where I can convince everyone it would be best to remove them, leaving only drone shells
> 
> PLUS, if that doesn't help any, the modules the drones get are pretty basic- there's a lot of things they just Can't do that cyborg modules can

So, I'd like to preface that I am actually hugely in favor of these additions. I think having more variety is ALWAYS a good thing, especially for AI / Cyborg players. Drones are cool and cute, and much like you said, not every play has a power fantasy of being optimized, sometimes it's nice to play something with less abilities. So I am more than okay with adding these, in fact, I'd like for drones to not differentiate between AI and cyborg players because functionally they are silicons and silicons should always be interchangeable.

That being said... the modules are too strong. I think it's okay for them to have modules, but it should be more like emergency stuff. Like, 4-5 items per module. Something like:

```
Civilian:
    - Extinguisher
    - Pitcher
    - Pen
    - Spray bottle
    - Sponge

Engineering:
    - Extinguisher
    - Welding tool
    - T-ray scanner
    - Device analyzer
    - Lamp manufacturer

Medsci:
    - Brute automender
    - Burn automender
    - Robotic hypospray
    - Syringe
    - Beaker (small)
```
That way they could fix issues when absolutely necessary, but they wouldn't be the dedicated person for it. They can communicate with departments and other cyborgs for that.

Now, to comment on your anti-bioshell stance... I think that's a silly take. Saying "AI gets to do another job on top of being AI" is silly, because many times the AI doesn't really *have* a job. Unless there's doors to be opened or criminals to monitor, being an AI means a lot of downtime. Worse still, while not in a shell, you cannot hear what people are saying locally, meaning you cannot involve yourself in RP. You can hear radio, but only some RP happens on radio. So that's why AI players spend a lot of their time into a shell, because they want to be part of the round, and they can't properly do that.

To further debunk the stance, why is "getting to do another job on top of being your primary job" a bad thing? It's a fundamental principle of many jobs. The Chief Engineer is both a Head and an Engineer. That means they have to do administrative stuff AND Engineering stuff. Heck, the captain is quite literally all of the jobs, whenever he wants to. Having a silicon variant of that is very much in the spirit of the game.

**Comment by Paaiii** [comment]
> That being said... the modules are too strong. I think it's okay for them to have modules, but it should be more like emergency stuff. Like, 4-5 items per module. Something like:

okay yeah that's a very fair point.. i can kill some of the big things for the modules for sure!

**Comment by Paaiii** [comment]
> Now, to comment on your anti-bioshell stance... I think that's a silly take. Saying "AI gets to do another job on top of being AI" is silly, because many times the AI doesn't really _have_ a job. Unless there's doors to be opened or criminals to monitor, being an AI means a lot of downtime. Worse still, while not in a shell, you cannot hear what people are saying locally, meaning you cannot involve yourself in RP. You can hear radio, but only some RP happens on radio. So that's why AI players spend a lot of their time into a shell, because they want to be part of the round, and they can't properly do that.

And-- I don't know! I play AI mostly on lowpop (10-20 people) and I always find myself pretty busy-- not neccecarily with opening doors and tracking criminals, but with comms and talking to people!
The AI has intercoms to hear locally, and when those aren't available, sure, I can see it, but the AI is an incredibly busy job if you're working to be active with it- It's a matter of seeing The Comms as your job. I get a huge amount of interaction just by staying active on general and department comms and always commenting on whatever's being said. It's a policy of 'everything on comms is directed At Me, and I should respond to it best I can.'
And if all else fails, I will point out that you can move your core. I didn't PR all those fancy new casings for you to rust in a high-security enclosure yanno

Either way, you don't need cyborg shells in order to use shells, you have drone shells for that. I just think that cyborg shells unencourage communications between the AI and cyborgs, and if the AI had DRONE shells, but not full CYBORG shells, that wouldn't be an issue somuch-- either way, it's outside the scope of this PR and is a topic for somewhere else hdsjksdg

**Comment by CheffieGithub** [comment]
I don't really think `/mob/living/silicon/adrone` is a good path especially when `/mob/living/silicon/drone` exists. Since the later is unused (and this seems to replace them in function somewhat) you could replace that, I haven't really got any ideas for alternative names otherwise though.

**Comment by Paaiii** [comment]
> I don't really think `/mob/living/silicon/adrone` is a good path especially when `/mob/living/silicon/drone` exists. Since the later is unused (and this seems to replace them in function somewhat) you could replace that, I haven't really got any ideas for alternative names otherwise though.

I'll do that yeah! HONESTLY I was just scared to go deleting ostensibly unused things left and right in case I broke something horribly without realizing it hdsjkdsgh
but adrone is a bit confusing so

**Comment by 444Portal** [comment]
Personally unsure about the Hypospray, Burn Mender and Mining Equipment, definitely depends on how you see them being used, though, if you get to select the module on first deploy then yeah, I'd be for removing the items I mentioned. Other than that though, oh my god I love all of this.

**Comment by Paaiii** [comment]
> I like the rest of the pr and I think its good so I wont be commenting on that, the boxbots are cute I enjoy em. Personally I dislike eyebots having modules in general, I think it ruins the thing that makes eyebots interesting (which is how limited they are). With how limited eyebots are, it makes their usage limited to simple tasks (like fixing hacked doors), or setting up more complex stuff.
> 
> I'm not against expanding what eyebots can do, but instead of making them into mediocre AI borgs, simple things that expand their uses. For a few things I think would be good, giving them an intercom so you can hide them in a locker and grab audio from an area, and giving them a gps so you can use them to find telesci offsets more easily.

I've drastically shrunk the scope of what those modules have in according with the feedback I've been getting- I agree! I think more eyebot functionality like those things you outlined would be nice.. but also this is a thing a player can be confined to now so I want to give it some degree of versatility..
I think things like shells working as intercoms would be super interesting, but might be best saved for a later PR hdsjksdg this one already does a fair bit

**Comment by DisturbHerb** [comment]
The volume of feedback received on this PR alone necessitates its own feedback thread on the forum, hence the label.

**Review by Kyle2143 — CHANGES_REQUESTED** [review]
Remove this feature:
4. Drones can accept brains as well as AI interfaces.

Drones have always been a way for AI to interact with the station and not to be an alternative to being a cyborg. They should not be a cyborg-lite that you can throw a human brain into.

**Review by mordent-goonstation — CHANGES_REQUESTED** [review]
Not looked at the code for this, in terms of the features I'd like the discussion in the thread resolved before a dev merges so flagging as such.

My concerns re: not having non-AI control drones stand at this time, and I'd still like the screen cosmetic thing in a separate PR as I view it as a separate chunk.

**Comment by Paaiii** [comment]
drone reskin option cut out for a seperate PR, AI controlled borgs removed bc people keep saying I might as well put it in the extant PR, drones no longer accept brains - will update description

**Comment by TobleroneSwordfish** [comment]
I can't get this PR to launch right now, a bunch of runtimes from `/datum/lifeprocess/robot_statusupdate/var/robot_owner` being null

**Comment by Paaiii** [comment]
hm that was not. the intention. whoops.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
General Stuff review (code review to follow)
- Atropine should not go in hyposprays, especially not auto-refilling eyebot ones.
- The light colour changing is cute, but currently adds a floating pink blob when the eyebot is turned off
image
- I'm not sure about this, but they feel noticeably faster than they did before? Eyebot movespeed was a mess so I wouldn't be surprised.
- Opening the interface should probably be done with a screwdriver instead of a crowbar to match cyborgs.
- Help messages for construction would be very cool 🥺
- There should be some kind of feedback on why you can't remove the default module from some eyebots, either grey out the option on the UI or add an error message like when you try to swap them.
- You might want to split the map changes onto a separate PR or the conflicts are going to start going exponential.
- The hats are incredibly cute.

File: code/modules/robotics/robot/robot_docking_station.dm [hunk 1]
```diff
@@ -974,7 +1071,7 @@ TYPEINFO(/obj/machinery/recharge_station)
 		if("cell-install")
 			if (!isrobot(src.occupant))
 				return
-			if (user == src.occupant)
+			if (user == src.occupant && !isdrone(user))
```
Comment (mordent-goonstation) [inline code/modules/robotics/robot/robot_docking_station.dm:1074]: Does this mean drones *can* switch out their own cell? 🤔

File: code/modules/tgui/states/default.dm [hunk 2]
```diff
@@ -40,16 +40,13 @@ var/global/datum/ui_state/tgui_default_state/tgui_default_state = new /datum/ui_
...
+	if(. < UI_INTERACTIVE)
 		return
 
-	// Robots can interact with anything they can see.
-	if(GET_DIST(src, src_object) <= ((WIDE_TILE_WIDTH - 1)/ 2))
-		return UI_INTERACTIVE
-
-	return UI_UPDATE // AI eyebots can receive updates from anything that the AI can see.
+	// Drones can interact with anything the AI can.
+	return UI_INTERACTIVE
```
Comment (mordent-goonstation) [inline code/modules/tgui/states/default.dm:49]: This is a quite important change, should be atomized into its own PR.

File: code/obj/item/mob_parts/robot_parts.dm [hunk 3]
```diff
@@ -1216,6 +1212,257 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts/leg/right)
...
 	syndicate = TRUE
 
+
+
+/obj/item/parts/robot_parts/drone_frame
+	name = "drone frame"
+	icon_state = "frame"
+	max_health = 5000
+	/// This will make the borg a syndie one
+	var/emagged = 0
```
Comment (mordent-goonstation) [inline code/obj/item/mob_parts/robot_parts.dm:1222]: No longer relevant, right?

File: code/obj/item/mob_parts/robot_parts.dm [hunk 4]
```diff
@@ -1216,6 +1212,257 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts/leg/right)
...
+	var/freemodule = TRUE
+	var/build_step = 0
+	var/shelltypetoapply = "eyebot"
+	var/moduletoapply = null
+	var/obj/item/cell/cell = null
+	var/obj/item/ai_interface/ai_interface = null
+	var/obj/item/ = null
```
Comment (mordent-goonstation) [inline code/obj/item/mob_parts/robot_parts.dm:1229]: ?

File: code/obj/item/mob_parts/robot_parts.dm [hunk 5]
```diff
@@ -1216,6 +1212,257 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts/leg/right)
...
+					return
+				else
+					boutput(user, "You need at least three lengths of cable to install it in [src].")
+					return
+			else if (src.build_step > 1)
+				boutput(user, "\The [src] already has wiring!")
+
+			else
+				boutput(user, "You need to plate the frame before you can add wires to it!")
+				return
```
Comment (mordent-goonstation) [inline code/obj/item/mob_parts/robot_parts.dm:1284]: Prefer putting "fail" cases first, then just doing the behavior once they've been checked and early-returned from.

File: code/obj/item/mob_parts/robot_parts.dm [hunk 6]
```diff
@@ -1216,6 +1212,257 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts/leg/right)
...
+
+		if (istype(W, /obj/item/device/radio))
+			if (src.build_step == 2)
+				var/obj/item/device/radio/M = W
```
Comment (mordent-goonstation) [inline code/obj/item/mob_parts/robot_parts.dm:1288]: `M` to something else.

File: code/obj/item/mob_parts/robot_parts.dm [hunk 7]
```diff
@@ -1216,6 +1212,257 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts/leg/right)
...
+
+		if (iswrenchingtool(W))
+			var/list/actions = list("Do nothing")
+			if(src.check_completion())
+				actions.Add("Finish and Activate the Drone")
+			if(src.ai_interface)
+				actions.Add("Remove the AI Interface")
+			if(src.cell)
+				actions.Add("Remove the Power cell")
+			if(!actions.len)
```
Comment (mordent-goonstation) [inline code/obj/item/mob_parts/robot_parts.dm:1348]:
```suggestion
			if(length(actions) == 0)
```

File: code/obj/item/mob_parts/robot_parts.dm [hunk 8]
```diff
@@ -1216,6 +1212,257 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts/leg/right)
...
+			src.UpdateOverlays(image('icons/mob/hivebot.dmi', "cell-c-" + src.shelltypetoapply, FLOAT_LAYER, 4),"cell")
+		else
+			src.UpdateOverlays(null,"cell")
+
+
+	proc/check_completion()
+		if (build_step == 3)
+			if ((src.ai_interface) && src.cell)
+				return 1
+		return 0
```
Comment (mordent-goonstation) [inline code/obj/item/mob_parts/robot_parts.dm:1409]:
```suggestion
				return TRUE
		return FALSE
```

File: code/obj/item/mob_parts/robot_parts.dm [hunk 9]
```diff
@@ -1216,6 +1212,257 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts/leg/right)
...
+
+		if (src.ai_interface)
+			if (!(drone in available_ai_shells))
+				available_ai_shells += drone
+			for_by_tcl(AI, /mob/living/silicon/ai)
+				boutput(AI, "<span class='success'>[src] has been connected to you as a controllable drone.</span>")
+			playsound(src, 'sound/weapons/radxbow.ogg', 40, 1)
+			drone.ai_interface = src.ai_interface
+			drone.ai_interface.set_loc(drone)
+			drone.shell = 1
```
Comment (mordent-goonstation) [inline code/obj/item/mob_parts/robot_parts.dm:1426]:
```suggestion
			drone.shell = TRUE
```
Broadly, check for usages of `1` that should be changed to `TRUE`, and usages of `0` that should be changed to `FALSE`.

File: code/obj/item/mob_parts/robot_parts.dm [hunk 10]
```diff
@@ -1216,6 +1212,257 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts/leg/right)
...
+
+	proc/check_completion()
+		if (build_step == 3)
+			if ((src.ai_interface) && src.cell)
+				return 1
+		return 0
+
+	proc/finish_drone()
+		var/mob/living/silicon/drone/drone = null
+		drone = new /mob/living/silicon/drone(get_turf(src.loc),src,0,src.emagged)
```
Comment (mordent-goonstation) [inline code/obj/item/mob_parts/robot_parts.dm:1413]: Can probably combine these together.

File: code/obj/item/mob_parts/robot_parts.dm [hunk 11]
```diff
@@ -1216,6 +1212,257 @@ ABSTRACT_TYPE(/obj/item/parts/robot_parts/leg/right)
...
+		qdel(src)
+		return
+
+/obj/item/parts/robot_parts/drone_frame/complete_shell_choosable
+	build_step = 3
+	cell = new/obj/item/cell/shell_cell/charged
+	ai_interface = new/obj/item/ai_interface
+	New()
+		..()
+		src.finish_drone()
```
Comment (mordent-goonstation) [inline code/obj/item/mob_parts/robot_parts.dm:1454]: If these are immediately turning into drones, what's the use case for this where you wouldn't just instantiate a drone?

File: code/obj/machinery/floorflusher.dm [hunk 12]
```diff
@@ -112,7 +112,7 @@ ADMIN_INTERACT_PROCS(/obj/machinery/floorflusher, proc/flush)
 					return
 				else if (G.active_tool == I)
 					return
-			else if (isrobot(user) || isshell(user)) // neither of these guys should be able to drop things in here!!
+			else if (isrobot(user)) // neither of these guys should be able to drop things in here!!
```
Comment (mordent-goonstation) [inline code/obj/machinery/floorflusher.dm:115]:
```suggestion
			else if (isrobot(user)) // these guys should not be able to drop things in here!!
```

**Review by mordent-goonstation — CHANGES_REQUESTED** [review]
I didn't review changes to .dmi or .dmm files, those will need looking over prior to merge.

Broad points:

- Still a change or two in here that are not within the scope of this PR. I called them out with comments, please do those in separate/follow-up PRs (and remove from this one). This is already a huge changeset, extra little "bonus" changes make it much harder to review.
- The `isrobot`, `iscyborg`, and `isdrone` macros are frustrating because the type of a cyborg is `.../robot`. The various checks for them that seem *relatively* random as to who can do what should probably go live somewhere else (shared) rather than how it currently is. Not part of this PR, should be a (soon to follow) follow-up change.
- As general feedback, this PR as a "make eyebots better" rework is solid, but really could have been split into several smaller ones that are going to make useful feedback much easier. Something to consider for future work/other people looking at this.

Please update the PR description to reflect the actual changes (and the changelog part, I guess) and let me know if you have any questions. Please ping me on Discord to review follow-up commits on this branch to try to get it looked at in a timely fashion.

Thanks!

File: code/modules/robotics/robot/robot_docking_station.dm (hunk 1 above)
Comment (Paaiii) [reply code/modules/robotics/robot/robot_docking_station.dm:1074]: Huh. No idea why that's there, no. Might have been leftover from before I realized AI eyes being able to change cells was a bug? Either way.

File: code/mob/living/critter/humanoid/brullbar.dm [hunk 13]
```diff
@@ -179,7 +179,7 @@
 		return ..() && (!frenzy.disabled && !fadeout.disabled) // so they can't attack you while frenzying or while invisible (kinda)
 
 	proc/fuck_up_silicons(var/mob/living/silicon/silicon) // modified orginal object critter behaviour scream
-		if (isrobot(silicon) && !ON_COOLDOWN(src, "brullbar_messup_cyborg", 30 SECONDS))
+		if (iscyborg(silicon) && !ON_COOLDOWN(src, "brullbar_messup_cyborg", 30 SECONDS))
```
[the remark this answers is not in the document]
Comment (Paaiii) [reply code/mob/living/critter/humanoid/brullbar.dm:182]: Eyebots shoooould be covered down at the else statement below? It does a good chunk random brute to a noncyborg silicon that gets hit with this proc, so.

File: code/modules/robotics/robot/module_tool_creator/construction.dm [hunk 14]
```diff
@@ -43,6 +43,11 @@
 	amount = 200
 	item_type = /obj/item/sheet/steel
 
+/datum/robot/module_tool_creator/item_type/amount/steel_sheet/drone
+	name = "steel sheets"
+	amount = 0
```
[the remark this answers is not in the document]
Comment (Paaiii) [reply code/modules/robotics/robot/module_tool_creator/construction.dm:48]: Yep, more or less!

File: code/modules/robotics/robot/robot_docking_station.dm [hunk 15]
```diff
@@ -525,24 +525,46 @@ TYPEINFO(/obj/machinery/recharge_station)
...
-		occupant_data["health"] = H.health
-		occupant_data["max_health"] = H.max_health
+	if (isdrone(src.occupant)) // drone handling
+		var/mob/living/silicon/drone/D = src.occupant
+		occupant_data["name"] = D.name
+		occupant_data["kind"] = "drone"
+		if (D.ai_interface)
+			occupant_data["user"] = "ai"
+		else
+			occupant_data["user"] = "unknown"
```
[the remark this answers is not in the document]
Comment (Paaiii) [reply code/modules/robotics/robot/robot_docking_station.dm:535]: It's for if the drone somehow has... NOT an AI interface in it. Probably leftover from when they could have brains in them.

File: code/obj/item/mob_parts/robot_parts.dm (hunk 4 above)
Comment (Paaiii) [reply code/obj/item/mob_parts/robot_parts.dm:1229]: Huh. Yeah. Question mark, I have no idea what that is, either. Let me see, I can probably get rid of it.

File: code/modules/robotics/robot/robot_docking_station.dm [hunk 16]
```diff
@@ -818,15 +870,60 @@ TYPEINFO(/obj/machinery/recharge_station)
...
+			. = TRUE
+
 		if("module-install")
 			if (!isrobot(src.occupant))
 				return
 			var/mob/living/silicon/robot/R = src.occupant
 			var/moduleRef = params["moduleRef"]
 			if(moduleRef)
 				var/obj/item/robot_module/module = locate(moduleRef) in src.modules
+				if (isdrone(src.occupant) && module.moduletype != "drone")
```
[the remark this answers is not in the document]
Comment (Paaiii) [reply code/modules/robotics/robot/robot_docking_station.dm:915]: Hmm.. It would probably make things cleaner if it did, yeah. A lot of things are shared between the two types which would be a lot easier to work with to if they were stored like that? I think.
I'm not the most experienced programmer, so take what I'm saying with a grain of salt, hwhaha.

File: code/obj/item/mob_parts/robot_parts.dm (hunk 11 above)
Comment (Paaiii) [reply code/obj/item/mob_parts/robot_parts.dm:1459]: I tried that at first! Creating the drone directly had some difficulties with... interface boards being unremovable, the HUD not showing up, a few other things... I couldn't figure out why for the life of me, so I pulled this as a workaround because it didn't cause those same issues.
I could try taking a further look at it, but I really have not the slightest what was going wrong with it, hwhaha.

File: code/mob/living/silicon/ai.dm [hunk 17]
```diff
@@ -1862,17 +1867,17 @@ or don't if it uses a custom topopen overlay
...
-	for (var/mob/living/silicon/hivebot/H in available_ai_shells)
-		if (H.shell && !H.dependent && !isdead(H))
-			bodies += H
-
 	for (var/mob/living/silicon/robot/R in available_ai_shells)
 		if (R.shell && !R.dependent && !isdead(R))
 			bodies += R
 
+	for (var/mob/living/silicon/drone/D in available_ai_shells)
+		if (D.shell && !D.dependent && !isdead(D))
```
[the remark this answers is not in the document]
Comment (Paaiii) [reply code/mob/living/silicon/ai.dm:1880]: I could have at the time- but given the recent update to AI cross zlevel interaction making cyborg shells undeployable on other zlevels, but drones not, this check actually requires the distinction between now..

**Comment by mordent-goonstation** [comment]
Good to look at again, or still WIP? (Not pressuring, just running through the various PRs I'm trying to keep track of.)

**Comment by Paaiii** [comment]
Good to look at, more or less
I need to fix the conflicts and diagnose the failed check but I've been busy so that's probably a bit a ways off
everything apart from that is in order so probably good to look at
